my_game.py

```diff
@@ -165,21 +169,28 @@ def on_update(self, delta_time):
         Movement and game logic
         """
 
-        # Calculate player speed based on the keys pressed
+        # Player has no speed unless buttons are pressed
         self.player.change_x = 0
+        self.player.change_y = 0
 
-        # Move player with keyboard
+        # Set speed for player with keyboard
         if self.left_pressed and not self.right_pressed:
-            self.player.change_x = -PLAYER_SPEED_X
+            self.player.change_x = -PLAYER_SPEED
         elif self.right_pressed and not self.left_pressed:
-            self.player.change_x = PLAYER_SPEED_X
+            self.player.change_x = PLAYER_SPEED
+        elif self.up_pressed and not self.down_pressed:
+            self.player.change_y = PLAYER_SPEED
+        elif self.down_pressed and not self.up_pressed:
+            self.player.change_y = -PLAYER_SPEED
 
         # Move player with joystick if present
         if self.joystick:
-            self.player.change_x = round(self.joystick.x) * PLAYER_SPEED_X
+            self.player.change_x = round(self.joystick.x) * PLAYER_SPEED
 
-        # Update player sprite
-        self.player.on_update(delta_time)
```

Comment:
Perhaps we could try adding back 
`self.player.on_update`

Reply:
> Perhaps we could try adding back `self.player.on_update`

Updating is now done by the physics engine (PE). We run the [_PE.update()_ function](https://api.arcade.academy/en/latest/_modules/arcade/physics_engines.html#PhysicsEngineSimple.update). That, in turn, calls __move_sprite()_ to manipulate the sprite's position.  You can read the source code [here](https://api.arcade.academy/en/latest/_modules/arcade/physics_engines.html).

So, no update function on the _Sprite_ is in use.

Reply:
ah, i see that - that's very good